### PR TITLE
fixing lock release race condition

### LINKED ISSUE #18: undefined method `-' for nil:NilClass
In the current HEAD version, I get `undefined method - for nil:NilClass` when calling

    self.with_lock timeout: 1.minute, wait: true, reload: true do

My backtrace is:

    /Users/me/.rvm/gems/ruby-1.9.3-p392/bundler/gems/mongoid-locker-19e6fccd5e38/lib/mongoid/locker.rb:143:in `lock'
    /Users/me/.rvm/gems/ruby-1.9.3-p392/bundler/gems/mongoid-locker-19e6fccd5e38/lib/mongoid/locker.rb:79:in `with_lock'
    /Users/me/Documents/Projects/priceapi_api/app/models/job.rb:89:in `handle_result'

This is the line

    retry_sleep = locked_until - Time.now

Which means, that `locked_until` is `nil` here.

### PR DESCRIPTION
**Pull request by pschrammel**
should fix #18

### CODE REVIEW

**Comment by dblock** [comment]
Makes sense. Please update CHANGELOG and fix the build (Rubocop). I'm cleaning up Rubocop configuration now so it will make things easier.

**Comment by afeld** [comment]
:+1:

**Comment by dblock** [comment]
Pull from upstream/rebase.

**Comment by pschrammel** [comment]
everthing done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## HEAD ([diff](https://github.com/afeld/mongoid-locker/compare/v0.3.0...master?w=1))
 
-* your contribution here
```
Comment (dblock) [inline CHANGELOG.md]: Out this back, below.

File: spec/mongoid-locker_spec.rb [hunk 2]
```diff
@@ -180,6 +180,18 @@ class User
       }.to raise_error(Mongoid::Locker::LockError)
     end
 
+    it 'should not fail if the lock has been released between check and sleep time calculation' do
```
Comment (dblock) [inline spec/mongoid-locker_spec.rb]: should not -> does not

**Comment by dblock** [comment]
Looks good. Can you please squash your commits? Thanks.

**Comment by pschrammel** [comment]
hmm ... somehow it was closed... I'll open a new one.

**Pull request closed without merging**